lib/asciidoctor-bibtex/extensions.rb

```diff
@@ -2,10 +2,10 @@
 # Treeprocessor extension for asciidoctor
 #
 
-require 'asciidoctor'
-require 'asciidoctor/extensions'
-require 'asciidoctor/reader'
-require 'asciidoctor/parser'
+require 'asciidoctor' unless RUBY_PLATFORM == 'opal'
+require 'asciidoctor/extensions' unless RUBY_PLATFORM == 'opal'
+require 'asciidoctor/reader' unless RUBY_PLATFORM == 'opal'
+require 'asciidoctor/parser' unless RUBY_PLATFORM == 'opal'
```

Comment:
you can use a block:

```suggestion
unless RUBY_PLATFORM == 'opal'
  require 'asciidoctor'
  require 'asciidoctor/extensions'
  require 'asciidoctor/reader'
  require 'asciidoctor/parser'
end
```